azure_functions_worker/bindings/datumdef.py

```diff
@@ -92,6 +94,55 @@ def from_typed_data(cls, td: protos.TypedData):
 
         return cls(val, tt)
 
+    @classmethod
+    def from_rpc_shared_memory(cls, shmem: protos.RpcSharedMemory, shmem_mgr: SharedMemoryManager) -> Optional[Datum]:
+        """
+        Reads the specified shared memory region and converts the read data into a datum object of
+        the corresponding type.
+        """
+        mmap_name = shmem.name
+        offset = shmem.offset
+        count = shmem.count
+        data_type = shmem.type
+        if data_type == protos.RpcDataType.bytes:
+            val = shmem_mgr.get_bytes(mmap_name, offset, count)
+            if val is not None:
+                return cls(val, 'bytes')
+        elif data_type == protos.RpcDataType.string:
+            val = shmem_mgr.get_string(mmap_name, offset, count)
+            if val is not None:
+                return cls(val, 'string')
+        return None
+
+    @classmethod
+    def to_rpc_shared_memory(cls, datum: Datum, shmem_mgr: SharedMemoryManager) -> protos.RpcSharedMemory:
+        """
+        Writes the given value to shared memory and returns the corresponding RpcSharedMemory
+        object which can be sent back to the functions host over RPC.
+        """
+        if datum.type == 'bytes':
+            value = datum.value
+            map_name = shmem_mgr.put_bytes(value)
+            if map_name is not None:
```

Comment:
The else of this if would end up into a not implemented for a datum type while it is not the issue here.